perun-core/src/main/java/cz/metacentrum/perun/core/api/AttributesManager.java

```diff
@@ -665,8 +661,10 @@
 	 * @throws WrongAttributeValueException if the attribute value is illegal
 	 * @throws WrongAttributeAssignmentException if attribute is not vo attribute
 	 * @throws WrongReferenceAttributeValueException
+	 * @throws GroupResourceMismatchException
+	 * @throws MemberResourceMismatchException
 	 */
-	void setAttributes(PerunSession sess, Vo vo, List<Attribute> attributes) throws PrivilegeException, InternalErrorException, VoNotExistsException, AttributeNotExistsException, WrongAttributeValueException, WrongAttributeAssignmentException, WrongReferenceAttributeValueException;
+	void setAttributes(PerunSession sess, Vo vo, List<Attribute> attributes) throws PrivilegeException, InternalErrorException, VoNotExistsException, AttributeNotExistsException, WrongAttributeValueException, WrongAttributeAssignmentException, WrongReferenceAttributeValueException, GroupResourceMismatchException, MemberResourceMismatchException;
```

Comment:
Such methods shouldn't throw ``GroupResourceMismatchException`` and ``MemberResourceMismatchException``, because Group/Resource/Member is not method param. It seems, that in most of methods, it's caused by ``checkAttributeDependencies()`` called in ``AttributesManagerBlImpl.`` Please catch exception in this method since is uses data retrieved inside the logic and they should be correct. If not, wrap it to ``ConsistencyErrorException`` or ``InternalErrorException``. This should lower the span of new exceptions in most methods. Please make it as a new pull-request, I will review the rest of changes ofter this is fixed. Thank you.